src/pass/llvmgen.cpp

```diff
@@ -392,6 +393,18 @@ Value* LLVMGen::visit(Store& store)
     return CreateStore(val, addr);
 }
 
+void LLVMGen::visit(AtomicAdd& add)
+{
+#ifdef ENABLE_CUDA
+    auto addr = eval(add.addr);
+    auto val = eval(add.val);
+    builder()->CreateAtomicRMW(AtomicRMWInst::Add, addr, val, MaybeAlign(),
```

Comment:
it looks like this implementation can be generalized in two ways.
1. CreateAtomicRMW supports many operations. not just Add. https://llvm.org/docs/LangRef.html#atomicrmw-instruction. we should change this to support all operations in MathOp.
2. AtomicRMW instruction is not specific to CUDA. it can be used for all backends. We should have generic support for atomic operations across all backends.

Reply:
I would suggest to define a statement called Atomic (or something), that takes a MathOp, an address expr and a value expression.